README.md

```diff
@@ -104,6 +103,46 @@ The following Visual Studio Code settings are available for the Nim extension.
 }
 ```
 
+#### Nim Lanugage Server integration (experimental)
+
+`vscode-nim` provides optional integration with [Nim Language Server](https://github.com/Nim-lang/langserver) as an
+alternative of using `nimsuggest`. In order to activate it, put the following
+lines in your config:
+
+```json
+{
+    "nim.enableLsp": true
+}
+```
+
+Similar to `vscode-nim` [Nim Language Server](https://github.com/Nim-lang/langserver) uses `nimsuggest` under the
+hood and to control the way `nimsuggest` is created you can use
+`nim.rootConfig`. Here it is a sample config:
+
+``` json
+{
+    "nim.enableLsp": true,
+    "nim.rootConfig": [{
+        // open files under tests using one nimsuggest instance started with root = test/all.nim
+        "root": "tests/all.nim",
+        "regexps": [
+            "tests/.*\\.nim"
+        ]
+    }, {
+        // everything else - use main.nim as root.
+        "root": "main.nim",
+        "regexps": [
+            ".*\\.nim"
+        ]
+    }]
+}
+```
+
+* (Re)starting `nimsuggest` subprocess
+
+Use `Command palette...`  -> `Source actions` -> `Restart nimsuggest`
```

Comment:
This should be under commands and the restart should work for either mode.

Reply:
This is a standard language server protocol approach and it will work fine across all clients without the need to create custom commands in all clients.

Reply:
I meant the headings commands in the readme.

Also, that it should look at the config, if there is epc nimsuggest it should restart those or of it's nimls it should restart it

Reply:
> Also, that it should look at the config, if there is epc nimsuggest it should restart those or of it's nimls it should restart it

The command comes as a result of `textDocument/codeActions` and it is implemented on the server thus it does not have access to the client state. I cannot think of an easy way to make restart work for both.

Reply:
https://code.visualstudio.com/api/extension-guides/command#registering-a-command

You can register a listener, it'll listen to the the command, just guard it on the config